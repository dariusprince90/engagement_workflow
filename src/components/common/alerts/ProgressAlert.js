import { memo } from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  alertType: PropTypes.oneOf(['info', 'warning']).isRequired,
  message: PropTypes.string.isRequired,
  percentComplete: PropTypes.number
};

let ProgressAlert = ({ alertType, message, percentComplete = 100 }) => {
  // **********************************************************************
  // * constants

  // **********************************************************************
  // * component vars

  // **********************************************************************
  // * functions

  // **********************************************************************
  // * side effects

  // **********************************************************************
  // * render

  return (
    <div className={`alert alert-${alertType} border border-${alertType}`}>
      <h5>{message}</h5>
      <div className={`progress rounded border border-${alertType}`}>
        <div
          className={`progress-bar progress-bar-striped progress-bar-animated bg-${alertType}`}
          style={{ width: `${percentComplete}%` }}
        />
      </div>
    </div>
  );
};

ProgressAlert = memo(ProgressAlert);
ProgressAlert.propTypes = propTypes;
ProgressAlert.displayName = 'ProgressAlert';

export default ProgressAlert;
