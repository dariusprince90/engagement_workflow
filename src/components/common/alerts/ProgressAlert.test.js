import ReactDOM from 'react-dom';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import faker from '@faker-js/faker';

import ProgressAlert from './ProgressAlert';

// **********************************************************************
// * constants

const defaultProps = {
  alertType: faker.helpers.arrayElement(['info', 'warning']),
  message: faker.datatype.string()
};

// **********************************************************************
// * functions

const getComponentToRender = (props) => {
  return <ProgressAlert {...props} />;
};

// **********************************************************************
// * mock external dependencies

// **********************************************************************
// * unit tests

describe('ProgressAlert', () => {
  // **********************************************************************
  // * setup

  // **********************************************************************
  // * tear-down

  // **********************************************************************
  // * execution

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(getComponentToRender(defaultProps), div);
  });

  it.each([{ alertType: 'info' }, { alertType: 'warning' }])(
    'sets the alert-, border-, and progress bar bg- classes to $alertType when the alertType is $alertType',
    ({ alertType }) => {
      const message = 'fixed-message';
      const props = { ...defaultProps, alertType, message };
      const tree = renderer.create(getComponentToRender(props)).toJSON();
      expect(tree).toMatchSnapshot();
    }
  );

  it('renders the message properly', () => {
    const props = { ...defaultProps, message: faker.datatype.string() };
    render(getComponentToRender(props));
    expect(screen.getByText(props.message)).toBeInTheDocument();
  });
});
