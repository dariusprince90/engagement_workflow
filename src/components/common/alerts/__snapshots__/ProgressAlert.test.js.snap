// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressAlert sets the alert-, border-, and progress bar bg- classes to info when the alertType is info 1`] = `
<div
  className="alert alert-info border border-info"
>
  <h5>
    fixed-message
  </h5>
  <div
    className="progress rounded border border-info"
  >
    <div
      className="progress-bar progress-bar-striped progress-bar-animated bg-info"
      style={
        Object {
          "width": "100%",
        }
      }
    />
  </div>
</div>
`;

exports[`ProgressAlert sets the alert-, border-, and progress bar bg- classes to warning when the alertType is warning 1`] = `
<div
  className="alert alert-warning border border-warning"
>
  <h5>
    fixed-message
  </h5>
  <div
    className="progress rounded border border-warning"
  >
    <div
      className="progress-bar progress-bar-striped progress-bar-animated bg-warning"
      style={
        Object {
          "width": "100%",
        }
      }
    />
  </div>
</div>
`;
